src/network/networkd-address.c

```diff
@@ -978,7 +981,7 @@ int config_parse_address_scope(const char *unit,
 bool address_is_ready(const Address *a) {
         assert(a);
 
-        if (a->family == AF_INET6)
+        if (a->family == AF_INET6 || a->preferred_lifetime_is_zero)
```

Comment:
How does `preferred_lifetime_is_zero` differ from `IFA_F_DEPRECATED`? Aren't they the same thing?
Also this never changed post-v239 so what was it that introduced the bug?

Reply:
When `PreferredLifetime=0`, the address has `IFA_F_DRPRECATED`, and it is expected. So, networkd considered that the address is ready even if the flag is set.

Without this change, the address cannot be ready, and thus, no consequent setting up process does not start. This is why you get the link state is in `configuring` rather than `configured`.

Reply:
> Also this never changed post-v239 so what was it that introduced the bug?

I guess 6aa5773bfff0a92d64da70426cae833df6f84daf introduced the bug.

Reply:
Apologies if I'm missing something here

> When `PreferredLifetime=0`, the address has `IFA_F_DRPRECATED`

If this is the case, can the function just be reduced to  `return !(a->flags & IFA_F_TENTATIVE);`? Or is there a case where an address is deprecated but `PreferredLifetime!=0`? I feel like the logic is duplicated

Reply:
Thank you for the comment. Updated. PTAL.

Reply:
Looks good, I hope it's correct. Please don't take my word for it though, I'm not entirely sure how it is supposed to work. There was probably a reason the IFA_F_DEPRECATED check was used originally

Reply:
Just tried and it seems possible to add routes to a deprecated address, so I'm not sure why it was even there in the first place :man_shrugging: 
I'll try this out tomorrow in the testing environment and let you know how it goes. Thanks very much :)